File: SDKs/drchrono.com/v4 (Hunt Valley)/typescript/src/sdk/models/operations/billingprofilesread.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class BillingProfilesReadPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class BillingProfilesReadQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=doctor" })
  doctor?: number;
}


export class BillingProfilesReadSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  drchronoOauth2: shared.SchemeDrchronoOauth2;
}


export class BillingProfilesReadRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: BillingProfilesReadPathParams;

  @SpeakeasyMetadata()
  queryParams: BillingProfilesReadQueryParams;

  @SpeakeasyMetadata()
  security: BillingProfilesReadSecurity;
}


export class BillingProfilesReadResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  billingProfile?: shared.BillingProfile;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/drchrono.com/v4 (Hunt Valley)/python/src/sdk/models/operations/billing_profiles_read.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import billingprofile as shared_billingprofile


@dataclasses.dataclass
class BillingProfilesReadPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class BillingProfilesReadQueryParams:
    doctor: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'doctor', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class BillingProfilesReadSecurity:
    drchrono_oauth2: shared_security.SchemeDrchronoOauth2 = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class BillingProfilesReadRequest:
    path_params: BillingProfilesReadPathParams = dataclasses.field()
    query_params: BillingProfilesReadQueryParams = dataclasses.field()
    security: BillingProfilesReadSecurity = dataclasses.field()
    

@dataclasses.dataclass
class BillingProfilesReadResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    billing_profile: Optional[shared_billingprofile.BillingProfile] = dataclasses.field(default=None)
    
